### PR TITLE
Add spec item for transportParams overriding default params

### PR DESCRIPTION
**Pull request by SimonWoolf**
See discussion https://ably-real-time.slack.com/archives/C030APSH3/p1503940782000117

### CODE REVIEW

**Comment by mattheworiordan** [comment]
@SimonWoolf thanks, looks good, mind rebasing on `master-1-1` which is now rebased on `master`?

File: content/client-lib-development-guide/features.textile [hunk 1]
```diff
@@ -271,6 +271,7 @@ h3(#realtimeclient). RealtimeClient
 ** @(RTC1d)@ @realtimeHost@ string, when set, will modify the realtime endpoint host used by this client library
 ** @(RTC1e)@ @environment@ string, when set, will modify both the REST and realtime endpoint hosts by prefixing the environment to the default endpoint host with a hypen delimiter. For example, a @RealtimeClient@ with an @environment@ of "sandbox", would use "sandbox-rest.ably.io" as the @restHost@ and @sandbox-realtime.ably.io@ as the @realtimeHost@. See "TO3k3":#TO3k3 for constraints.
 ** @(RTC1f)@ @transportParams@ map or equivalent, additional parameters to be sent in the querystring when initiating a realtime connection. Keys are @Strings@, values are @Stringifiable@ (a value that can be coerced to a string in order to be sent as a querystring parameter. Supported values should be at least strings, numbers, and booleans, with booleans stringified as @true@ and @false@. If this is unidiomatic to the language, the implementer may consider this as equivalent to @String@).
+*** @(RTC1f1)@ If a key in @transportParams@ is one the library sends by default (for example, @v@ or @heartbeats@), the value in @transportParams@ takes precendence.
```
Comment (paddybyers) [inline content/client-lib-development-guide/features.textile]: Typo "precendence" -> "precedence"

**Comment by paddybyers** [comment]
LGTM apart from comment

**Comment by SimonWoolf** [comment]
fixed typo & rebased

**Pull request merged**